USERMANUAL.md

```diff
@@ -0,0 +1,599 @@
+3. resolves the `entity_uris` defined in all `Rule Binding` to retrieve the corresponding table and column names for validation
+4. generate BigQuery SQL code for each `Rule Binding` and validate that it is valid BigQuery SQL using BigQuery dry-run feature
+5. create a BigQuery view for each generated BigQuery SQL `Rule Binding` in the BigQuery dataset specified in `--gcp_bq_dataset_id`
+6. create a BigQuery job to execute all `Rule Binding` SQL views. The BigQuery jobs will be created in the GCP project specified in `--gcp_project_id` and GCP region specified in `--gcp_region_id`
+7. aggregate the validation outcomes and write the Data Quality summary results into a table called `dq_summary` in the BigQuery dataset specified in `--gcp_bq_dataset_id`
+8. if the CLI argument `--target_bigquery_summary_table` is defined, CloudDQ will append all rows created in the `dq_summary` table in the last run to the BigQuery table specified in  `--target_bigquery_summary_table`. `--target_bigquery_summary_table` currently an optional argument to maintain backwards compatibility but will become a required argument in version 1.0.0.
+9. if the CLI flag `--summary_to_stdout` is set, CloudDQ will retrieve all validation results rows created in the `dq_summary` table in the last run and log them as a JSON record to Cloud Logging and stdout
+
+To execute all `Rule Binding`s discovered in the config directory, use `ALL` as the `RULE_BINDING_IDS`:
+
+```bash
+#!/bin/bash
+export GOOGLE_CLOUD_PROJECT=$(gcloud config get-value project)
+export CLOUDDQ_BIGQUERY_REGION=EU
+export CLOUDDQ_BIGQUERY_DATASET=clouddq
+export CLOUDDQ_TARGET_BIGQUERY_TABLE="<project-id>.<dataset-id>.<table-id>"
+python3 clouddq_executable.zip \
+    ALL \
+    configs \
+    --gcp_project_id="${GOOGLE_CLOUD_PROJECT}" \
+    --gcp_bq_dataset_id="${CLOUDDQ_BIGQUERY_DATASET}" \
+    --gcp_region_id="${CLOUDDQ_BIGQUERY_REGION}" \
+    --target_bigquery_summary_table="${CLOUDDQ_TARGET_BIGQUERY_TABLE}"
+```
+
+To see the resulting Data Quality validation summary statistics in the BigQuery table `$CLOUDDQ_TARGET_BIGQUERY_TABLE`, run:
+```bash
+#!/bin/bash
+export GOOGLE_CLOUD_PROJECT=$(gcloud config get-value project)
+export CLOUDDQ_BIGQUERY_REGION=EU
+export CLOUDDQ_TARGET_BIGQUERY_TABLE="<project-id>.<dataset-id>.<table-id>"
+echo "select * from \`${CLOUDDQ_TARGET_BIGQUERY_TABLE}\`" | bq query --location=${CLOUDDQ_BIGQUERY_REGION} --nouse_legacy_sql
+```
+
+To see the BigQuery SQL logic generated by CloudDQ for each `Rule Binding` stored as a view in `--gcp_bq_dataset_id`, run:
+```bash
+#!/bin/bash
+export CLOUDDQ_BIGQUERY_DATASET=clouddq
+bq show --view ${CLOUDDQ_BIGQUERY_DATASET}.T3_DQ_1_EMAIL_DUPLICATE
+```
+
+
+
+## Example Use Cases
+
+In this section, we will provide some example implementations of data quality checks that are common requirements.
+
+### Aggregate Conditions
+
+#### Unique Values Constraint
+
+Ensures that all values in a column are unique.
+
+```yaml
+rules:
+  UNIQUE:
```

Comment:
pls rewrite this rule (and the similar rule below) as a CUSTOM_SQL_EXPRESSION rule

Reply:
I don't think uniqueness can be done as a CUSTOM_SQL_EXPRESSION...?

Reply:
I dropped the "aggregate conditions" section that has these rules, and added the examples to the "set-based validations" section.